ironfist/src/cpp/combat/combat_spells.cpp

```diff
@@ -137,4 +138,25 @@ void combatManager::Resurrect(int spell, int hex, int spellpower) {
 	}
 	this->DrawFrame(1, 0, 0, 0, 75, 1, 1);
 	creat->creature.creature_flags &= ~DEAD;
-}
\ No newline at end of file
+}
+
+float army::SpellCastWorkChance(int spell)
+{
+	if (this->effectStrengths[EFFECT_ANTI_MAGIC]

```

Comment:
And's bind tighter than or's, but you've written them at the same indentation, so it looks like they'll equal. It's hard to tell what exactly the condition is when this is true.

But really, you'll only modifying this in the case where this is an earth elemental. This can be shortened to:

if (this->effectStrengths[EFFECT_ANTI_MAGIC]) {
  return 0.0;
}